BitwardenShared/UI/Auth/StartRegistration/StartRegistrationProcessor.swift

```diff
@@ -0,0 +1,237 @@
+            guard !email.isEmpty else {
+                throw StartRegistrationError.emailEmpty
+            }
+
+            guard email.isValidEmail else {
+                throw StartRegistrationError.invalidEmail
+            }
+
+            coordinator.showLoadingOverlay(title: Localizations.creatingAccount)
+
+            let result = try await services.accountAPIService.startRegistration(
+                requestModel: StartRegistrationRequestModel(
+                    email: email,
+                    name: name,
+                    receiveMarketingEmails: state.isReceiveMarketingToggleOn
+                )
+            )
+
+            if let token = result.token,
+               !token.isEmpty {
+                coordinator.navigate(to: .completeRegistration(
+                    emailVerificationToken: token,
+                    userEmail: state.emailText
+                ))
+            } else {
+                coordinator.navigate(to: .checkEmail(email: state.emailText))
+            }
+        } catch let error as StartRegistrationError {
+            showStartRegistrationErrorAlert(error)
+        } catch {
+            coordinator.showAlert(.networkResponseError(error) {
+                await self.startRegistration()
+            })
+        }
+    }
+
+    /// Sets the region to the last used region.
+    ///
+    private func loadRegion() async {
+        guard let urls = await services.stateService.getPreAuthEnvironmentUrls() else {
+            await setRegion(.unitedStates, urls: .defaultUS)
+            return
+        }
+
+        if urls.base == EnvironmentUrlData.defaultUS.base {
+            await setRegion(.unitedStates, urls: urls)
+        } else if urls.base == EnvironmentUrlData.defaultEU.base {
+            await setRegion(.europe, urls: urls)
+        } else {
+            await setRegion(.selfHosted, urls: urls)
+        }
+    }
+
+    /// Builds an alert for region selection and navigates to the alert.
+    ///
+    private func presentRegionSelectionAlert() {
+        let actions = RegionType.allCases.map { region in
+            AlertAction(title: region.baseUrlDescription, style: .default) { [weak self] _ in
+                if let urls = region.defaultURLs {
+                    await self?.setRegion(region, urls: urls)
+                } else {
+                    self?.coordinator.navigate(
+                        to: .selfHosted(currentRegion: self?.state.region ?? .unitedStates),
+                        context: self
+                    )
+                }
+            }
+        }
+        let cancelAction = AlertAction(title: Localizations.cancel, style: .cancel)
+        let alert = Alert(
+            title: Localizations.loggingInOn,
+            message: nil,
+            preferredStyle: .actionSheet,
+            alertActions: actions + [cancelAction]
+        )
+        coordinator.showAlert(alert)
+    }
```

Comment:
🎨 Can these be extracted into a helper class that can be shared with LandingProcessor?